narwhals/functions.py

```diff
@@ -594,17 +599,19 @@ def read_csv(
     impl = Implementation.from_backend(backend)
     native_namespace = impl.to_native_namespace()
     native_frame: NativeFrame
-    if impl in {
-        Implementation.POLARS,
-        Implementation.PANDAS,
-        Implementation.MODIN,
-        Implementation.CUDF,
-    }:
-        native_frame = native_namespace.read_csv(source, **kwargs)
+    if impl in {Implementation.PANDAS, Implementation.MODIN, Implementation.CUDF}:
+        native_frame = native_namespace.read_csv(source, sep=separator, **kwargs)
+    elif impl is Implementation.POLARS:
+        native_frame = native_namespace.read_csv(source, separator=separator, **kwargs)
     elif impl is Implementation.PYARROW:
+        if separator is not None and "parse_options" in kwargs:
+            msg = "Can't pass both `separator` and `parse_options`."
+            raise TypeError(msg)
         from pyarrow import csv  # ignore-banned-import
 
-        native_frame = csv.read_csv(source, **kwargs)
+        native_frame = csv.read_csv(
+            source, parse_options=csv.ParseOptions(delimiter=separator), **kwargs
+        )
```

Comment:
I think this is a bit odd:

1. `separator` is not typed to be `None`
2. Even if that was the case, the following would not error in line 607: 
    ```py
    nw.read_csv(..., separator=None, parse_option=csv.ParseOptions(...), backend=nw.Implementation.PYARROW)
    ```
    However, then in line 613, we would call
    ```py
    csv.read_csv(
            source, parse_options=csv.ParseOptions(delimiter=None), parse_options=parse_options, ...
        )
    ```
    which would end up raising an exception at this point

3. Should we handle the same for other backends? i.e. pandas-like check that `sep` is not passed, and below for lazy backends